### ISSUE TITLE
dpctl.tensor.logaddexp does not always return `nan` when one operand is not `nan`

### ISSUE THREAD

**Issue by ndgrigorian** [IntelPython/dpctl#1321]
For example:
```
In [1]: import dpctl.tensor as dpt

In [2]: x1, x2 = dpt.asarray(1., dtype=dpt.float64), dpt.asarray(dpt.nan, dtype=dpt.float64)

In [3]: dpt.logaddexp(x1, x2)
Out[3]: usm_ndarray(1.69314718)

In [4]: dpt.logaddexp(x2, x1)
Out[4]: usm_ndarray(nan)
```
This should return `nan` in both cases

**Comment by ndgrigorian** [comment]
Another case:
```
In [3]: x1 = dpt.asarray(dpt.nan)
x
In [4]: x1 = dpt.asarray(dpt.inf)

In [5]: x2 = dpt.asarray(dpt.inf)

In [6]: dpt.logaddexp(x1, x2)
Out[6]: usm_ndarray(nan)
```

**Comment by ndgrigorian** [comment]
This issue was resolved in #1324
